lib/code_corps/sparkpost/emails/receipt.ex

```diff
@@ -0,0 +1,81 @@
+defmodule CodeCorps.SparkPost.Emails.Receipt do
+
+  alias CodeCorps.{
+    DonationGoal,
+    Project,
+    Repo,
+    StripeConnectCharge,
+    StripeConnectSubscription,
+    User,
+    WebClient
+  }
+  alias SparkPost.{Content, Transmission}
+
+  @high_five_image_urls [
+    "https://d3pgew4wbk2vb1.cloudfront.net/emails/images/emoji-1f64c-1f3fb@2x.png",
+    "https://d3pgew4wbk2vb1.cloudfront.net/emails/images/emoji-1f64c-1f3fc@2x.png",
+    "https://d3pgew4wbk2vb1.cloudfront.net/emails/images/emoji-1f64c-1f3fd@2x.png",
+    "https://d3pgew4wbk2vb1.cloudfront.net/emails/images/emoji-1f64c-1f3fe@2x.png",
+    "https://d3pgew4wbk2vb1.cloudfront.net/emails/images/emoji-1f64c-1f3ff@2x.png"
+  ]
+
+  @spec build(StripeConnectCharge.t, Stripe.Invoice.t) :: %Transmission{}
+  def build(%StripeConnectCharge{} = charge, %Stripe.Invoice{} = invoice) do
+    with %StripeConnectCharge{user: %User{} = user} = charge <- Repo.preload(charge, :user),
+         %Project{} = project <- invoice.subscription |> get_project(),
+         {:ok, %DonationGoal{} = current_donation_goal} <- project |> get_current_donation_goal()
+    do
```

Comment:
The email building can fail. I think this should go to the event handler. Data should be fetched/preloaded there.